### PR TITLE
stabilise mixed_integer_ops

### LINKED ISSUE #87840: Tracking Issue for `#![feature(mixed_integer_ops)]`
Feature gate: `#![feature(mixed_integer_ops)]`

### Public API

```rust
// `uX` is `u8`, `u16`, `u32`, `u64`,`u128`, `usize`
impl uX {
    pub const fn checked_add_signed(self, iX) -> Option<Self>;
    pub const fn overflowing_add_signed(self, iX) -> (Self, bool);
    pub const fn saturating_add_signed(self, iX) -> Self;
    pub const fn wrapping_add_signed(self, iX) -> Self;
}

impl iX {
    pub const fn checked_add_unsigned(self, uX) -> Option<Self>;
    pub const fn overflowing_add_unsigned(self, uX) -> (Self, bool);
    pub const fn saturating_add_unsigned(self, uX) -> Self;
    pub const fn wrapping_add_unsigned(self, uX) -> Self;

    pub const fn checked_sub_unsigned(self, uX) -> Option<Self>;
    pub const fn overflowing_sub_unsigned(self, uX) -> (Self, bool);
    pub const fn saturating_sub_unsigned(self, uX) -> Self;
    pub const fn wrapping_sub_unsigned(self, uX) -> Self;
}
```

### Steps / History

- [x] Implementation: #87601 
- [ ] Final comment period (FCP)
- [ ] Stabilization PR

### PR DESCRIPTION
**Pull request by Dylan-DPC**
Closes #87840

### CODE REVIEW

**Comment by joshtriplett** [comment]
@rfcbot merge

**Comment by lukas-code** [comment]
From #87601:
> Maybe it would be interesting to also have `add_signed` that panics in debug and wraps in release ?

Has this been addressed yet? There doesn't seem to be much interest in that particular function tho.

**Comment by joshtriplett** [comment]
@Dylan-DPC Could you please update the `since` fields? r=me with those updated.

@bors delegate+

**Comment by est31** [comment]
:wave: Hello, I'm writing this comment in this **stabilization PR** to notify you, the authors of this PR, that https://github.com/rust-lang/rust/pull/100591 has been merged, which implemented a change in how features are stabilized.

Your PR has been filed before the change, so will likely require modifications in order to comply with the new rules. I recommend you to:

1. rebase the PR onto latest master, so that uses of the placeholder are possible.
2. replace the version numbers in the PR with the placeholder `CURRENT_RUSTC_VERSION`. For language changes, this means the version numbers in `accepted.rs` (example: 4caedbae869855a19ca49455e2cd338ae3b8c1ef). For library changes, this means the `since` fields (example e576a9b554f24a26c36c04b41c901636d190605b).

That's it! The `CURRENT_RUSTC_VERSION` placeholder will, as part of the release process, be replaced with the version number that the PR merged for. It can be used anywhere in `rust-lang/rust`, not just accepted.rs and the since fields.

If you have any questions, feel free to drop by the [zulip stream](https://rust-lang.zulipchat.com/#narrow/stream/241545-t-release/topic/libs.20stabilization.20placeholder), or ping me directly in this PR's thread. Thanks! :wave:

File: library/core/src/num/uint_macros.rs [hunk 1]
```diff
@@ -1051,14 +1050,13 @@ macro_rules! uint_impl {
...
         ///
         /// ```
-        /// # #![feature(mixed_integer_ops)]
         #[doc = concat!("assert_eq!(1", stringify!($SelfT), ".saturating_add_signed(2), 3);")]
         #[doc = concat!("assert_eq!(1", stringify!($SelfT), ".saturating_add_signed(-2), 0);")]
         #[doc = concat!("assert_eq!((", stringify!($SelfT), "::MAX - 2).saturating_add_signed(4), ", stringify!($SelfT), "::MAX);")]
         /// ```
-        #[unstable(feature = "mixed_integer_ops", issue = "87840")]
-        #[rustc_const_unstable(feature = "mixed_integer_ops", issue = "87840")]
-        #[must_use = "this returns the result of the operation, \
```
Comment (jhpratt) [inline library/core/src/num/uint_macros.rs:1061]: Looks like an accidental change on this line.

**Comment by Dylan-DPC** [comment]
closing this as #101555 is being merged

**Pull request closed without merging**